rule all:
    input:
        "trimmed_fastq/T0-L01-02_R1_paired.fq.gz",
        "trimmed_fastq/T0-L01-02_R1_unpaired.fq.gz",
        "trimmed_fastq/T0-L01-02_R2_paired.fq.gz",
        "trimmed_fastq/T0-L01-02_R2_unpaired.fq.gz",
        "kallisto_aligned/T0-L01-02"

rule trim_reads:
    input:
        "rna-seq_fastq/T0-L01-02_S1_L001_R1_001.fastq.gz",
        "rna-seq_fastq/T0-L01-02_S1_L001_R2_001.fastq.gz"
    output:
        "trimmed_fastq/T0-L01-02_R1_paired.fq.gz",
        "trimmed_fastq/T0-L01-02_R1_unpaired.fq.gz",
        "trimmed_fastq/T0-L01-02_R2_paired.fq.gz",
        "trimmed_fastq/T0-L01-02_R2_unpaired.fq.gz"
    shell:
        "java -jar ~/usr/bin/Trimmomatic-0.36/trimmomatic-0.36.jar PE -phred33 {input} {output} LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:35"

rule kallisto_quant:
    input:
        index="reference_data/transcripts.idx",
        r1="trimmed_fastq/T0-L01-02_R1_paired.fq.gz",
        r2="trimmed_fastq/T0-L01-02_R2_paired.fq.gz"
    output:
        directory("kallisto_aligned/T0-L01-02")
    shell:
        "kallisto quant -b 10  -i {input.index} -o {output} {input.r1} {input.r2}"
